src/main.rs

```diff
@@ -35,33 +35,32 @@ fn main() {
     let pattern = matches.value_of("pattern").unwrap_or(&empty_pattern);
 
     // Get the current working directory
-    let current_dir_buf = match env::current_dir() {
-        Ok(cd) => cd,
-        Err(_) => error("Error: could not get current directory."),
-    };
-    let current_dir = current_dir_buf.as_path();
+    let current_dir = Path::new(".");
+    if !current_dir.is_dir() {
+        error("Error: could not get current directory.");
+    }
 
     // Get the root directory for the search
-    let mut root_dir_is_absolute = false;
-    let root_dir_buf = if let Some(rd) = matches.value_of("path") {
-        let path = Path::new(rd);
-
-        root_dir_is_absolute = path.is_absolute();
-
-        fshelper::absolute_path(path).unwrap_or_else(|_| {
-            error(&format!("Error: could not find directory '{}'.", rd))
-        })
-    } else {
-        current_dir_buf.clone()
+    let mut root_dir_buf = match matches.value_of("path") {
+        Some(path) => PathBuf::from(path),
+        None => current_dir.to_path_buf(),
     };
-
     if !root_dir_buf.is_dir() {
         error(&format!(
             "Error: '{}' is not a directory.",
             root_dir_buf.to_string_lossy()
         ));
     }
 
+    let path_display = if matches.is_present("absolute-path") || root_dir_buf.is_absolute() {
+        PathDisplay::Absolute
+    } else {
+        PathDisplay::Relative
+    };
+
+    if path_display == PathDisplay::Absolute && root_dir_buf.is_relative() {
+        root_dir_buf = fshelper::absolute_path(root_dir_buf.as_path()).unwrap();
```

Comment:
We should avoid using `unwrap` here:
```
> fd -a non/existing/relative/path
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { repr: Os { code: 2, message: "No such file or directory" } }', /checkout/src/libcore/result.rs:906:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

Reply:
I think you forgot to add the pattern.

Reply:
You are absolutely right :smile:. So `unwrap` was only called because the current directory did not exist (the issue above).